core/src/test/java/io/jenkins/pluginhealth/scoring/scores/SecurityWarningScoringTest.java

```diff
@@ -45,19 +49,68 @@ SecurityWarningScoring getSpy() {
-            KnownSecurityVulnerabilityProbe.KEY, ProbeResult.success(KnownSecurityVulnerabilityProbe.KEY, "SECURITY-123, link-to-security-advisory", 1)
-        ));
+        when(plugin.getDetails())
+                .thenReturn(Map.of(
+                        KnownSecurityVulnerabilityProbe.KEY,
+                        ProbeResult.success(
+                                KnownSecurityVulnerabilityProbe.KEY, "SECURITY-123:link-to-security-advisory", 2)));
+
+        final ScoreResult result = scoring.apply(plugin);
+
+        assertThat(result.key()).isEqualTo("security");
+        assertThat(result.weight()).isEqualTo(1f);
+        assertThat(result.value()).isEqualTo(0);
+        result.componentsResults().forEach(ScoringComponentResult -> {
+            ScoringComponentResult.resolutions().forEach(resolution -> {
+                assertThat(resolution.link()).isEqualTo("link-to-security-advisory");
+                assertThat(resolution.text()).isEqualTo("SECURITY-123");
+            });
+        });
+    }
+
+    @Test
+    void shouldBeAbleToDetectPluginWithMultipleSecurityWarning() {
+        final Plugin plugin = mock(Plugin.class);
+        final SecurityWarningScoring scoring = getSpy();
+        final List<String> secruityInfo = new ArrayList<>();
+        secruityInfo.add("SECURITY-123:link-to-security-advisory");
+        secruityInfo.add("SECURITY-123:link-to-security-advisory");
+        when(plugin.getDetails())
+                .thenReturn(Map.of(
+                        KnownSecurityVulnerabilityProbe.KEY,
+                        ProbeResult.success(
+                                KnownSecurityVulnerabilityProbe.KEY,
+                                secruityInfo.get(0) + "," + secruityInfo.get(1),
+                                2)));
 
         final ScoreResult result = scoring.apply(plugin);
 
         assertThat(result.key()).isEqualTo("security");
         assertThat(result.weight()).isEqualTo(1f);
         assertThat(result.value()).isEqualTo(0);
+        final List<Resolution> resolutionList = new ArrayList<>();
+        secruityInfo.stream().forEach(securityDetail -> {
+            String[] res = securityDetail.split(",");
+            Arrays.stream(res).forEach(securityMessage -> {
+                String[] securityInfo = securityMessage.split(":");
+                resolutionList.add(new Resolution(securityInfo[0].trim(), securityInfo[1]));
+            });
+        });
+
+        result.componentsResults().forEach(ScoringComponentResult -> {
+            AtomicInteger count = new AtomicInteger();
+            ScoringComponentResult.resolutions().forEach(resolution -> {
+                assertThat(resolutionList.get(0).link())
+                        .isEqualTo(resolution.link().trim());
+                assertThat(resolutionList.get(1).text())
+                        .isEqualTo(resolution.text().trim());
+                count.incrementAndGet();
+            });
+        });
```

Comment:
You can simplify that using `assertThat(result.componentResults().contains()` (not the exact method be should put you close enough to find it).

Reply:
Tried to simply thanks @alecharp